shell/platform/common/accessibility_bridge.cc

```diff
@@ -524,7 +533,7 @@ void AccessibilityBridge::SetTreeData(const SemanticsNode& node,
   // 1. this text field has a valid selection
   // 2. this text field doesn't have a valid selection but had selection stored
   //    in the tree.
-  if (flags & FlutterSemanticsFlag::kFlutterSemanticsFlagIsTextField) {
+  if (flags & FlutterSemanticsFlag::kFlutterSemanticsFlagIsTextField && flags & FlutterSemanticsFlag::kFlutterSemanticsFlagIsFocused) {
```

Comment:
We want to only update these values for the focused node so that the text selection data points to the focused node, not to whichever text node happens to be latest in the update.